### PR TITLE
p2p: update state sync messages for reverse sync

### PR DESCRIPTION
**Pull request by cmwaters**
Adds `LightBlockRequest` and `LightBlockResponse` as per https://github.com/tendermint/spec/pull/224

### CODE REVIEW

File: spec/p2p/messages/state-sync.md [hunk 1]
```diff
@@ -70,16 +71,41 @@ possibly from a different peer.
 
 The ABCI application is able to request peer bans and chunk refetching as part of the ABCI protocol.
 
+### LightBlockRequest
```
Comment (marbar3778) [inline spec/p2p/messages/state-sync.md:74]: There is a proto directory in this repo as well. Could you add these there?

Comment (cmwaters) [reply spec/p2p/messages/state-sync.md:74]: Yup. I didn't see that 👍

**Comment by marbar3778** [comment]
@tendermint/informal-systems could we get an approval here.

File: proto/statesync/types.proto [hunk 2]
```diff
@@ -35,3 +39,11 @@ message ChunkResponse {
   bytes  chunk   = 4;
   bool   missing = 5;
 }
+
+message LightClientRequest {
```
Comment (milosevic) [inline proto/statesync/types.proto:43]:
```suggestion
message LightBlockRequest {
```

File: proto/statesync/types.proto [hunk 3]
```diff
@@ -35,3 +39,11 @@ message ChunkResponse {
...
+  uint64 height = 1;
+}
+
+message LightClientResponse {
```
Comment (milosevic) [inline proto/statesync/types.proto:47]:
```suggestion
message LightBlockResponse {
```

**Review by milosevic — APPROVED** [review]
LGTM. There are just minor typos in message names.

File: proto/statesync/types.proto (hunk 2 above)
Comment (cmwaters) [reply proto/statesync/types.proto:43]: Thanks for that!

**Pull request merged**